content/blog/2021/09/camel-kafka-consumer-changes/index.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "Upcoming changes to the consumer on Kafka component"
+date: 2021-09-27
+draft: true
+authors: [orpiske]
+categories: ["Features", "Camel"]
+preview: "Upcoming changes to the consumer on Kafka component"
+---
+
+We recently started to review, investigate and fix reliability issues with the [Kafka](https://camel.apache.org/components/latest/kafka-component.html) component. As part of that, we refactored the code to avoid problematic concurrency patterns, remove calls to deprecated Kafka client code and simplify the overall maintainability of the code. This led to many changes in the internal implementation of the component, which we will talk about in this blog post.
+
+## Motivation
+
+The ground work related to these changes can be traced back to the following set of issues reported on our Jira. Some of these issues were relatively new, but some have been reported for some time. In no particular order, we have:
+
+* [camel-kafka: possible corruption of idempotency messages when using KafkaIdempotentRepository](https://issues.apache.org/jira/browse/CAMEL-16914)
+* [camel-kafka: usage of deprecated method may cause Camel to block indefinitely](https://issues.apache.org/jira/browse/CAMEL-16928)
+* [camel-kafka: investigate alternatives to remove Thread.sleep from the record fetcher](https://issues.apache.org/jira/browse/CAMEL-16949)
+* [camel-kafka: commits may be skipped on stop or shutdown](https://issues.apache.org/jira/browse/CAMEL-16973)
+* [camel-kafka: improve the documentation about consumer streams vs consumers count](https://issues.apache.org/jira/browse/CAMEL-16981)
+* [camel-kafka -consumerCount vs consumerStreams problem with pool size](https://issues.apache.org/jira/browse/CAMEL-13215)
+* [camel-kafka: make the resume strategy configurable](https://issues.apache.org/jira/browse/CAMEL-16974)
+* [Seek to specific offset and KafkaConsumer access](https://issues.apache.org/jira/browse/CAMEL-13768)
```

Comment:
would look nicer if this also starts with camel-kafka:

Reply:
+1, thanks! I'll adjust the text here and on the ticket as well.